File: feature/star_trek/src/main/java/com/example/lego/feature/startrek/StarTrekViewModel.kt
package com.example.lego.feature.startrek

import androidx.lifecycle.ViewModel
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow

data class STCharacter(
    val id: String,
    val name: String,
    val rank: String,
    val actor: String
)

@HiltViewModel
class StarTrekViewModel @Inject constructor() : ViewModel() {

    private val _characters = MutableStateFlow<List<STCharacter>>(emptyList())
    val characters: StateFlow<List<STCharacter>> = _characters

    init {
        _characters.value = listOf(
            STCharacter("1", "James T. Kirk", "Captain", "William Shatner"),
            STCharacter("2", "Spock", "Commander", "Leonard Nimoy"),
            STCharacter("3", "Leonard McCoy", "Lieutenant Commander", "DeForest Kelley"),
            STCharacter("4", "Nyota Uhura", "Lieutenant", "Nichelle Nichols"),
            STCharacter("5", "Hikaru Sulu", "Lieutenant", "George Takei")
        )
    }

    fun getCharacterById(id: String): STCharacter? {
        return _characters.value.find { it.id == id }
    }
}

File: Sources/Features/StarTrek/StarTrekViewModel.swift
import SwiftUI

struct STCharacter: Identifiable {
    let id: String
    let name: String
    let rank: String
    let actor: String
}

class StarTrekViewModel: ObservableObject {
    @Published var characters: [STCharacter] = []
    
    init() {
        characters = [
            STCharacter(id: "1", name: "James T. Kirk", rank: "Captain", actor: "William Shatner"),
            STCharacter(id: "2", name: "Spock", rank: "Commander", actor: "Leonard Nimoy"),
            STCharacter(id: "3", name: "Leonard McCoy", rank: "Lieutenant Commander", actor: "DeForest Kelley"),
            STCharacter(id: "4", name: "Nyota Uhura", rank: "Lieutenant", actor: "Nichelle Nichols"),
            STCharacter(id: "5", name: "Hikaru Sulu", rank: "Lieutenant", actor: "George Takei")
        ]
    }
    
    func getCharacterById(id: String) -> STCharacter? {
        characters.first { $0.id == id }
    }
}
